core/src/main/java/org/apache/carbondata/core/datamap/dev/DataMapWriter.java

```diff
@@ -45,14 +45,13 @@
    * @param blockletId sequence number of blocklet in the block
    */
   void onBlockletEnd(int blockletId);
-
   /**
    * Add the column pages row to the datamap, order of pages is same as `indexColumns` in
    * DataMapMeta returned in DataMapFactory.
    *
    * Implementation should copy the content of `pages` as needed, because `pages` memory
    * may be freed after this method returns, if using unsafe column page.
    */
-  void onPageAdded(int blockletId, int pageId, ColumnPage[] pages);
+  void onPageAdded(int blockletId, int pageId, ColumnPage[] pages, String directoryPath);
```

Comment:
instead of passing the directoryPath for every call, better add init method and pass the segment directory path to it.

Reply:
This is not a directory path, instead the full path to the block. SO this is constant for a block. So removed it from OnPageAdded and passed as a extra parameter to OnBlockStart as this blockPath has to be initialized on every block start.

Reply:
why do we require block path?